File: frontend/mts-ui/src/app/service/user-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface UserRegistrationRequest {
  name: string;
  email: string;
  password: string;
}

export interface UserLoginRequest {
  email: string;
  password: string;
}

export interface UserLoginResponse {
  accountId: number;
  holderName: string;
  message: string;
}

@Injectable({
  providedIn: 'root',
})
export class UserService {
  private readonly BASE_URL = 'http://localhost:8080/api/v1/users';

  constructor(private http: HttpClient) {}

  register(request: UserRegistrationRequest): Observable<{ message: string }> {
    return this.http.post<{ message: string }>(`${this.BASE_URL}/register`, request);
  }

  login(request: UserLoginRequest): Observable<UserLoginResponse> {
    return this.http.post<UserLoginResponse>(`${this.BASE_URL}/login`, request);
  }
}

File: backend/MoneyTransferSystem/src/main/java/com/fidelity/mts/service/UserService.java
package com.fidelity.mts.service;

import com.fidelity.mts.application.dto.UserLoginRequest;
import com.fidelity.mts.application.dto.UserLoginResponse;
import com.fidelity.mts.application.dto.UserRegistrationRequest;

public interface UserService {

    /** Registers a new user and creates a linked account with balance 0. */
    String register(UserRegistrationRequest request);

    /** Authenticates a user by email/password, returns their accountId. */
    UserLoginResponse login(UserLoginRequest request);
}
